tests/perf/workflows/workflow_test.go

```diff
@@ -71,28 +69,42 @@ func TestMain(m *testing.M) {
 	os.Exit(tr.Start(m))
 }
 
-func runk6test(t *testing.T, config K6RunConfig) *loadtest.K6RunnerMetricsSummary {
-	k6Test := loadtest.NewK6(
-		"./test.js",
-		loadtest.WithParallelism(1),
-		// loadtest.EnableLog(), // uncomment this to enable k6 logs, this however breaks reporting, only for debugging.
-		loadtest.WithRunnerEnvVar("TARGET_URL", config.TARGET_URL),
-		loadtest.WithRunnerEnvVar("SCENARIO", config.SCENARIO),
-		loadtest.WithRunnerEnvVar("WORKFLOW_NAME", config.WORKFLOW_NAME),
-		loadtest.WithRunnerEnvVar("WORKFLOW_INPUT", config.WORKFLOW_INPUT),
-		loadtest.WithRunnerEnvVar("RATE_CHECK", config.RATE_CHECK),
-	)
-	defer k6Test.Dispose()
-	t.Log("running the k6 load test...")
-	require.NoError(t, tr.Platform.LoadTest(k6Test))
-	sm, err := loadtest.K6ResultDefault(k6Test)
-	require.NoError(t, err)
-	require.NotNil(t, sm)
-	bts, err := json.MarshalIndent(sm, "", " ")
-	require.NoError(t, err)
-	require.True(t, sm.Pass, fmt.Sprintf("test has not passed, results %s", string(bts)))
-	t.Logf("test summary `%s`", string(bts))
-	return sm.RunnersResults[0]
+func collect(t *testing.T, testAppName string, table *summary.Table) {
+	appUsage, err1 := tr.Platform.GetAppUsage(testAppName)
+	sidecarUsage, err2 := tr.Platform.GetSidecarUsage(testAppName)
+	if err1 == nil {
+		table.
+		Outputf(appendTime(testAppName+"App Memory"), "%vMb", appUsage.MemoryMb).
+		Outputf(appendTime(testAppName+"App CPU"), "%vm", appUsage.CPUm)
+	}
+	if err2 == nil {
+		table.
+		Outputf(appendTime(testAppName+"Sidecar Memory"), "%vMb", sidecarUsage.MemoryMb).
+		Outputf(appendTime(testAppName+"Sidecar CPU"), "%vm", sidecarUsage.CPUm)
+	}
```

Comment:
Do we want to fail, or at the very least log a message, for instances where we aren't able to collect data? i.e. when the errors are not null

Reply:
Added a log message when app/sidecar is not available. 
The reason I don't fail when errors happen there is that sometimes the pod takes time to be created, and we could try to collect usage data before app/sidecar pods are ready, but we don't want to fail the tests in this case. We will continue querying the usage until the test finished and use those metrics to help identify any issue like memory leak.